specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimdeletedservices.json

```diff
@@ -0,0 +1,121 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ApiManagementClient",
+    "description": "Use these REST APIs for performing operations in Azure API Management deployment.",
+    "version": "2020-06-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "subscriptions/{subscriptionId}/providers/Microsoft.ApiManagement/deletedservices": {
+      "get": {
+        "tags": [
+          "DeletedService"
+        ],
+        "operationId": "DeletedServices_ListBySubscription",
+        "description": "Lists all soft-deleted services available for undelete for the given subscription.",
+        "x-ms-examples": {
+          "ApiManagementListRegions": {
+            "$ref": "./examples/ApiManagementDeletedServicesListBySubscription.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of all soft-deleted services available for undelete for the given subscription.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/DeletedServicesCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/deletedservices/{serviceName}": {
```

Comment:
How are you going to route this request to the right region?  Please follow KeyVaults deletedVaults APIs for listing/getting soft deleted entities.  specification\keyvault\resource-manager\Microsoft.KeyVault\preview\2020-04-01-preview\keyvault.json

Reply:
our RP will know the region from subId/RG/servicename. Or I'm not getting the question.

Reply:
every region of your RP knows about every deletedservice from every other region?  Aren't they regional resources?  This is why keyvault has location based routing for the individual gets to deletedVaults, so only the region that the vault was originally in needs to know about the soft-deleted variant

Reply:
Just a single RP is registered with ARM, so all the calls go to it first (let's call this one Global RP). The Global RP knows about all the regional RPs and also knows in what region any particularservice resides (by subId/RG/serviceName). From there the Global RP routes the call to the regional RP.

Reply:
@pilor your comment totally makes sense once we setup our regional RPs with ARM. Addressed that in the latest commit.